Source/UI/Delegates/ThreedsWebViewControllerDelegate.swift

```diff
@@ -4,8 +4,23 @@ import Foundation
 public protocol ThreedsWebViewControllerDelegate: AnyObject {
 
     /// Called if the response is successful
+    @available(*, deprecated, renamed: "threeDSWebViewControllerAuthenticationDidSucceed(_:token:)")
     func onSuccess3D()
 
     /// Called if the response is unsuccesful
+    @available(*, deprecated, renamed: "threeDSWebViewControllerAuthenticationDidFail(_:)")
     func onFailure3D()
+
+    /// Called upon successful 3D Secure authentication.
+    /// - Parameters:
+    ///   - threeDSWebViewController: The `ThreedsWebViewController` instance calling this method.
+    ///   - token: The token extracted from the success URL.
+    func threeDSWebViewControllerAuthenticationDidSucceed(_ threeDSWebViewController: ThreedsWebViewController,
+                                                          token: String?)
+
+    /// Called upon unsuccessful 3D Secure authentication.
+    /// - Parameters:
+    ///   - threeDSWebViewController: The `ThreedsWebViewController` instance calling this method.
+    func threeDSWebViewControllerAuthenticationDidFail(_ threeDSWebViewController: ThreedsWebViewController)
```

Comment:
Bit unsure on why we're returning the threeDSWebViewController here. Should we keep it or remove it?

Reply:
Good question if we are already on threeDSWebViewController we would already have its instance here. If we have a example using threeDSWebViewController then we may find a answer to this.

Reply:
I think he included it to follow other iOS delegate methods by Apple, e.g. `UITableViewDelegate` has `func tableView(_ tableView: UITableView, ...)`, and WK delegates, etc. I think we should keep it to stick with this standard and because the integrator might create an object that conforms to `ThreedsWebViewControllerDelegate`, but doesn't actually hold a reference to `ThreedsWebViewController`. These methods would give them access to that.